### DISCUSSION TITLE
usual bundlesize

### DISCUSSION THREAD

**Discussion by spacepolice10** [remix-run/remix#9517] [General]
Hi there,

Recently I built simple application and built it. I see there, that router is quite huge and makes bundle larget that it supposed to. 

<img width="947" alt="Screenshot 2024-05-30 at 22 02 03" src="https://github.com/remix-run/remix/assets/44314842/a05b5101-050d-4a8f-bebd-fa32031495c6">

Then I decided to use Rollup visualizer.

It shows me that tools like undici take up to 600kb (!) client-side. Is it something that is ok for Remix? Or should I reconsider configuration of the build process? How weighty your applications and what you usually do to make client-side lighter with Remix?

<img width="1173" alt="Screenshot 2024-05-30 at 22 03 49" src="https://github.com/remix-run/remix/assets/44314842/dd224d15-d598-4dbc-a8d2-a4b3db0f932b">

**Comment by sergiodxa** [comment]
Are you sure this is the client bundle? I see you have `@remix-run/node` and `@remix-run/server-runtime` which are server-only packages, same with Undici which is used only in Node and not in the browser.

Is this being tested in a new app?

**Reply by spacepolice10** [reply]
@sergiodxa yep, it is something that file `stats.html` in folder build/client shows. The app barely has some code, but have some vite plugins already installed. Feels like I accidentally brought the whole node_modules to the build without optimising it 😅

**Comment by spacepolice10** [comment]
Ok, I tried to use `vite-env-only` plugin to check out if some server code goes to client build and it seems to be like that. I have an error, but I don't understand why. It tries to bundle `createFileSessionStorage` even though I never use one in my app 
<img width="589" alt="Screenshot 2024-06-09 at 00 07 33" src="https://github.com/remix-run/remix/assets/44314842/eccfc077-27d6-40ea-85c3-09f41cde6636">

**Reply by spacepolice10** [reply]
@sergiodxa  how do you think, should it be considered a bug as if error originates from some inner components of either Remix or vite-env-only? P.S. If auto import is used VSCode randomly exports function from remix-run/node and remix-run/react. This one causes problem and Remix says nothing about that. Really hard to debug such things )
